R/aggregate.R

```diff
@@ -12,34 +12,39 @@
 #' @param second_weights A boolean indicating whether secondary weights should be used
+#'
+#' @examples
+#' staggregate_polynomial(demo_prcp, "prcp", daily_agg = "sum", degree = 3, geoweights_table = demo_output_geoweights, second_weights = TRUE)
+#' staggregate_spline(demo_prcp, "prcp", daily_agg = "sum", knot_locs = c(1,2,3), geoweights_table = demo_output_geoweights, second_weights = TRUE)
+#' staggregate_bin(demo_prcp, "prcp", daily_agg = "sum", num_bins = 30, min = 5, geoweights_table = demo_output_geoweights, second_weights = TRUE)
+#'
+#'
 
 ## Sara Orofino
 ## February 22, 2022
 ## Aggregate Climate Data: Pipeline Step 02
 
-#' @export
-agg_climate_data <- function(year, data_raster, climate_var, daily_agg, trans = 'polynomial', trans_specs, geoweights_table, second_weights = FALSE) {
-
-  # Function to convert raster to data.table from https://gist.github.com/etiennebr/9515738
-  as.data.table.raster <- function(x, row.names = NULL, optional = FALSE, xy=FALSE, inmem = raster::canProcessInMemory(x, 2), ...) {
-    if(inmem) {
-      v <- data.table::as.data.table(raster::as.data.frame(x, row.names=row.names, optional=optional, xy=xy, ...))
-    } else {
-      tr <- raster::blockSize(x, n=2)
-      l <- lapply(1:tr$n, function(i)
-        data.table::as.data.table(raster::as.data.frame(raster::getValues(x,
-                                                                  row=tr$row[i],
-                                                                  nrows=tr$nrows[i]),
-                                                        row.names=row.names, optional=optional, xy=xy, ...)))
-      v <- data.table::rbindlist(l)
-    }
-    coln <- names(x)
-    if(xy) coln <- c("x", "y", coln)
-    data.table::setnames(v, coln)
-    v
+# Function to convert raster to data.table from https://gist.github.com/etiennebr/9515738
+as.data.table.raster <- function(x, row.names = NULL, optional = FALSE, xy=FALSE, inmem = raster::canProcessInMemory(x, 2), ...) {
+  if(inmem) {
+    v <- data.table::as.data.table(raster::as.data.frame(x, row.names=row.names, optional=optional, xy=xy, ...))
+  } else {
+    tr <- raster::blockSize(x, n=2)
+    l <- lapply(1:tr$n, function(i)
+      data.table::as.data.table(raster::as.data.frame(raster::getValues(x,
+                                                                        row=tr$row[i],
+                                                                        nrows=tr$nrows[i]),
+                                                      row.names=row.names, optional=optional, xy=xy, ...)))
+    v <- data.table::rbindlist(l)
   }
-  #setMethod('as.data.table', signature(x='Raster'), as.data.table.raster)
+  coln <- names(x)
+  if(xy) coln <- c("x", "y", coln)
+  data.table::setnames(v, coln)
+  v
+}
 
+# Function to convert raster to data table and aggregate to daily values before transformation
+before_trans <- function(data_raster, climate_var, daily_agg, geoweights_table, second_weights = FALSE){
```

Comment:
why do you need `geoweights_table` as an argument here? just for the extent of the area to crop to from the global raster?

Reply:
Correct, I've removed it from before_trans' output so the setup list only contains two objects now (clim_daily and layer_names)